source/engine/engine/NcEngineImpl.h

```diff
@@ -19,6 +19,7 @@ namespace nc
             void Shutdown() noexcept override;
             auto GetComponentRegistry() noexcept -> ecs::ComponentRegistry& override;
             auto GetModuleRegistry() noexcept -> ModuleRegistry* override;
+            auto GetAsyncDispatcher() noexcept -> task::AsyncDispatcher override;
```

Comment:
Earlier in `ModuleFactory`, we pass the `AsyncDispatcher` as a const ref all the way down into `TaskRouter` until we copy it into a member variable at that point. Don't we want downstream callers of `GetAsyncDispatcher` to follow the same pattern? Why not const ref here?